internal/server/boltdbstate/project_test.go

```diff
@@ -0,0 +1,241 @@
+		// Set a config at the project and app scope
+		require.NoError(s.BuildPut(false, &pb.Build{
+			Id: "testBuild",
+			Application: &pb.Ref_Application{
+				Application: appName,
+				Project:     projectName,
+			},
+			Workspace: &pb.Ref_Workspace{Workspace: "default"},
+		}))
+
+		require.NoError(s.ArtifactPut(false, &pb.PushedArtifact{
+			Id: "testArtifact",
+			Application: &pb.Ref_Application{
+				Application: appName,
+				Project:     projectName,
+			},
+			Workspace: &pb.Ref_Workspace{Workspace: "default"},
+		}))
+
+		require.NoError(s.DeploymentPut(false, &pb.Deployment{
+			Id: "testDeployment",
+			Application: &pb.Ref_Application{
+				Application: appName,
+				Project:     projectName,
+			},
+			Workspace: &pb.Ref_Workspace{Workspace: "default"},
+		}))
+
+		require.NoError(s.ReleasePut(false, &pb.Release{
+			Id: "testRelease",
+			Application: &pb.Ref_Application{
+				Application: appName,
+				Project:     projectName,
+			},
+			Workspace: &pb.Ref_Workspace{Workspace: "default"},
+		}))
+
+		require.NoError(s.StatusReportPut(false, &pb.StatusReport{
+			Id: "testStatusReport",
+			Application: &pb.Ref_Application{
+				Application: appName,
+				Project:     projectName,
+			},
+			Workspace: &pb.Ref_Workspace{Workspace: "default"},
+		}))
+
+		require.NoError(s.ConfigSet(&pb.ConfigVar{
+			Target: &pb.ConfigVar_Target{
+				AppScope: &pb.ConfigVar_Target_Project{Project: &pb.Ref_Project{Project: projectName}},
+			},
+			Name:       "testProjectConfig",
+			Value:      &pb.ConfigVar_Static{Static: "paladin"},
+			Internal:   false,
+			NameIsPath: false,
+		}))
+
+		require.NoError(s.ConfigSet(&pb.ConfigVar{
+			Target: &pb.ConfigVar_Target{
+				AppScope: &pb.ConfigVar_Target_Application{Application: &pb.Ref_Application{
+					Project:     projectName,
+					Application: appName,
+				}},
+			},
+			Name:       "testAppConfig",
+			Value:      &pb.ConfigVar_Static{Static: "devops"},
+			Internal:   false,
+			NameIsPath: false,
+		}))
+
+		require.NoError(s.WorkspacePut(&pb.Workspace{
+			Name: "testWorkspace",
+			Projects: []*pb.Workspace_Project{
+				{
+					Project: &pb.Ref_Project{Project: projectName},
+				},
+			},
+		}))
+
+		require.NoError(s.PipelinePut(&pb.Pipeline{
+			Id:   "testPipeline",
+			Name: "testPipeline",
+			Owner: &pb.Pipeline_Project{
+				Project: &pb.Ref_Project{
+					Project: projectName,
+				},
+			},
+			Steps: map[string]*pb.Pipeline_Step{
+				"testStep": {
+					Name: "testStep",
+					Kind: &pb.Pipeline_Step_Up_{
+						Up: &pb.Pipeline_Step_Up{},
+					},
+				},
+			},
+		}))
+
+		require.NoError(s.TriggerPut(&pb.Trigger{
+			Id:        "testTrigger",
+			Name:      "testTrigger",
+			Project:   &pb.Ref_Project{Project: projectName},
+			Workspace: &pb.Ref_Workspace{Workspace: "testWorkspace"},
+		}))
+
+		// Delete the project (this should also delete the build)
+		err = s.ProjectDelete(&pb.Ref_Project{Project: projectName})
+		require.NoError(err)
```

Comment:
Small nit but you can combine a require.NoError/require.Error and a function so it's all on a single line:

```golang
require.NoError(s.ProjectDelete(&pb.Ref_Project{Project: projectName}))
```